File: 190521/sony/kth_number.py


def solution(array,commands):
    answer = []
    for i in range(len(commands)):
        start = commands[i][0]
        end = commands[i][1]
        kth = commands[i][2]
        a = array[start-1:end]
        a.sort()
        answer.append(a[kth-1])
    return answer

arr = [1, 5, 2, 6, 3, 7, 4]
commands = [[2, 5, 3], [4, 4, 1], [1, 7, 3]]
print(solution(arr,commands))



def solution(array, commands):
    answer = []
    for c in commands:
        s = c[0] - 1
        e = c[1]
        k = c[2] - 1
        answer.append(sorted(array[s:e])[k])
    return answer

File: 190521/hyodol/k-th_number.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

vector<int> solution(vector<int> array, vector<vector<int> > commands) {
    vector<int> answer;
    
    for (int i = 0; i < commands.size(); i++) {
        vector<int> tempArray = array;
        sort(tempArray.begin() + commands[i][0] - 1, tempArray.begin() + commands[i][1]);
        answer.push_back(tempArray[commands[i][0] + commands[i][2] - 2]);
    }
    
    return answer;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    
    vector<int> array = {1, 5, 2, 6, 3, 7, 4};
    vector<vector<int> > commands = {{2, 5, 3}, {4, 4, 1}, {1, 7, 3}};
    vector<int> answer = solution(array, commands);
    for (int num : answer) cout << num << " ";
    cout << '\n';
    
    return 0;
}

